octopoes/tools/xtdb-cli.py

```diff
@@ -0,0 +1,258 @@
+        headers = {"Accept": "application/json"}
+        req = httpx.get(self.root(f"entity?eid={key}"), headers=headers)
+        return req.text
+
+    def history(self, key: str):
+        headers = {"Accept": "application/json"}
+        req = httpx.get(self.root(f"entity?eid={key}&history=true&sortOrder=asc"), headers=headers)
+        return req.text
+
+    def entity_tx(self, key: str):
+        headers = {"Accept": "application/json"}
+        req = httpx.get(self.root(f"entity-tx?eid={key}"), headers=headers)
+        return req.text
+
+    def attribute_stats(self):
+        headers = {"Accept": "application/json"}
+        req = httpx.get(self.root("attribute-stats"), headers=headers)
+        return req.text
+
+    def sync(self, timeout: int = 500):
+        headers = {"Accept": "application/json"}
+        req = httpx.get(self.root(f"sync?timeout={timeout}"), headers=headers)
+        return req.text
+
+    def await_tx(self, txid: int):
+        headers = {"Accept": "application/json"}
+        req = httpx.get(self.root(f"await-tx?txId={txid}"), headers=headers)
+        return req.text
+
+    def await_tx_time(self, tm: str = datetime.datetime.now().isoformat()):
+        headers = {"Accept": "application/json"}
+        req = httpx.get(self.root(f"await-tx-time?tx-time={tm}"), headers=headers)
+        return req.text
+
+    def tx_log(self):
+        headers = {"Accept": "application/json"}
+        req = httpx.get(self.root("tx-log"), headers=headers)
+        return req.text
+
+    def submit_tx(self, txs):
+        headers = {"Accept": "application/json", "Content-Type": "application/json"}
+        data = '{{"tx-ops": {}}}'.format(" ".join(txs))
+        req = httpx.post(self.root("submit-tx"), headers=headers, data=data)
+        return req.text
+
+    def tx_committed(self, txid: int):
+        headers = {"Accept": "application/json"}
+        req = httpx.get(self.root(f"tx_commited?txId={txid}"), headers=headers)
+        return req.text
+
+    def latest_completed_tx(self):
+        headers = {"Accept": "application/json"}
+        req = httpx.get(self.root("latest-completed-tx"), headers=headers)
+        return req.text
+
+    def latest_submitted_tx(self):
+        headers = {"Accept": "application/json"}
+        req = httpx.get(self.root("latest-submitted-tx"), headers=headers)
+        return req.text
+
+    def active_queries(self):
+        headers = {"Accept": "application/json"}
+        req = httpx.get(self.root("active-queries"), headers=headers)
+        return req.text
+
+    def recent_queries(self):
+        headers = {"Accept": "application/json"}
+        req = httpx.get(self.root("recent-queries"), headers=headers)
+        return req.text
+
+    def slowest_queries(self):
+        headers = {"Accept": "application/json"}
+        req = httpx.get(self.root("recent-queries"), headers=headers)
+        return req.text
+
+
+def dispatch(xtdb, instruction):
+    match instruction.pop(0):
```

Comment:
```suggestion
    methodname = instruction.pop(0).replace('-', "_")
    try:
      method =  getattr(xtdb, methodname)
      if instruction:
        return method(instruction[0])
      return method()
    except AttributeError:
      match instruction.pop(0):
      # remaining specific calls.
      
```

Reply:
Addressed in 9efc251db6cd65359b7bb24febf57c6282479c66.